tests/test_composites.py

```diff
@@ -103,10 +103,102 @@ def test_composite_name_map_registered(name_map_composite, model_vision):
             for names, hook_template in name_map_composite.name_map:
                 if name in names:
                     if not check_hook_registered(child, hook_template):
-                        errors.append(
+                        errors.append((
+                            '{} is first in name map for {}, but is not registered!',
+                            (name, hook_template),
+                        ))
+                    break
+
+    if not verify_no_hooks(model_vision):
+        errors.append(('Model has hooks registered after composite was removed!', ()))
+
+    assert not errors, 'Errors:\n  ' + '\n  '.join(f'[{n}] ' + msg.format(*arg) for n, (msg, arg) in enumerate(errors))
+
+
+def test_composite_mixed_registered(mixed_composite, model_vision):
+    '''Tests whether the constructed MixedComposites register and unregister their rules correctly.'''
+    errors = []
+
+    name_map_composite = mixed_composite.composites[0]
+    special_first_layer_map_composite = mixed_composite.composites[1]
```

Comment:
Why not use unpacking here?
Then you also get an error if the `name_layer_map_composite` is unexpectedly longer than 2.

Reply:
fixed.